### PR TITLE
[results] Supports to Filter PipelineRun in List API

### PR DESCRIPTION
**Pull request by XinruZhang**
Updates the env settings for CEL(declares one more variable called `pipelinerun`), 
and endows the List Results API in `server.go` the ability to filter the results fetched 
from the database by either `taskrun` or `pipelinerun` or both.

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)
- [ ] Commit messages includes a project tag in the subject - e.g. [OCI], [hub], [results], [task-loops]

_See [the contribution guide](https://github.com/tektoncd/experimental/blob/master/CONTRIBUTING.md)
for more details._

### CODE REVIEW

File: results/pkg/api/server/server_test.go [hunk 1]
```diff
@@ -322,17 +342,29 @@ func TestListResults(t *testing.T) {
...
-			name:         "test simple query",
+			name:         "test query taskrun",
 			filter:       `taskrun.api_version=="v1beta1"`,
 			expect:       []*pb.Result{gotResults[0], gotResults[2], gotResults[3]},
 			expectStatus: codes.OK,
 		},
 		{
-			name:         "test query with simple function",
+			name:         "test query pipelinerun",
+			filter:       `pipelinerun.api_version=="v1beta1"`,
```
Comment (wlynch) [inline results/pkg/api/server/server_test.go:352]: Let's also test unions of mixed results - e.g. `taskrun.api_version=="v1beta1" || pipelinerun.api_version=="v1beta1"`

**Review by wlynch — COMMENTED** [review]
Looks great! One test case we should add, otherwise lgtm!

File: results/pkg/api/server/server_test.go (hunk 1 above)
Comment (XinruZhang) [reply results/pkg/api/server/server_test.go:352]: Great! Added :)

**Review by wlynch — APPROVED** [review]
/lgtm

**Comment by XinruZhang** [comment]
Had conflicts with the new master branch.

File: results/pkg/api/server/server.go [hunk 2]
```diff
@@ -189,22 +191,43 @@ func (s *Server) ListResultsResult(ctx context.Context, req *pb.ListResultsReque
...
 		}
 	}
 	return &pb.ListResultsResponse{Results: resp}, nil
 }
 
+// Check if the result can be reserved.
+func matchCelFilter(r *pb.Result, prg cel.Program) (bool, error) {
+	for _, e := range r.Executions {
+		var (
+			taskrun     pipeline_go_proto.TaskRun
```
Comment (wlynch) [inline results/pkg/api/server/server.go:207]: style: package names shouldn't have underscores (https://blog.golang.org/package-names). Consider ppb or pipelinepb instead.

File: results/pkg/api/server/server_test.go [hunk 3]
```diff
@@ -2,10 +2,13 @@ package server
 
 import (
 	"context"
+	"fmt"
 	"sort"
 	"testing"
 
+	"github.com/golang/protobuf/proto"
 	"github.com/google/go-cmp/cmp"
+	"github.com/tektoncd/experimental/results/proto/pipeline/v1beta1/pipeline_go_proto"
```
Comment (wlynch) [inline results/pkg/api/server/server_test.go:11]: This is a duplicate import with the package below.

**Comment by wlynch** [comment]
Also make sure to include a `[results]` project tag in the commit/pr title!

**Review by wlynch — APPROVED** [review]
/lgtm

**Pull request merged**